pkg/ipam/crd_eni.go

```diff
@@ -155,7 +160,7 @@ func waitForNetlinkDevices(configByMac configMap) (linkByMac linkMap, err error)
 	return linkByMac, errors.New("timed out waiting for ENIs to be attached")
 }
 
-func configureENINetlinkDevice(link netlink.Link, cfg eniDeviceConfig) error {
+func configureENINetlinkDevice(link netlink.Link, cfg eniDeviceConfig, useENIPrimaryIP bool) error {
```

Comment:
nit: The overall logic of the agent side of things looks sound. But you could add the `usePrimary` boolean to `eniDeviceConfig` instead. Or maybe just check if `cfg.ip` is not nil when setting the `cfp.ip` on the link.

Reply:
Good idea. That sounds cleaner. This change would need `parseENIConfig`'s function signature to be updated though. Will make this change.